components/event/org.wso2.carbon.event.admin/src/main/java/org/wso2/carbon/event/admin/internal/TopicManagerAdminService.java

```diff
@@ -82,56 +137,110 @@ public void updatePermission(String topic, TopicRolePermission[] topicRolePermis
+    @SuppressWarnings("UnusedDeclaration")
     public int getAllWSSubscriptionCountForTopic(String topic) throws EventAdminException {
         EventBroker eventBroker = EventAdminHolder.getInstance().getEventBroker();
         try {
             return eventBroker.getTopicManager().getSubscriptions(topic, true).length;
         } catch (EventBrokerException e) {
+            log.error("Error in accessing topic manager", e);
             throw new EventAdminException("Error in accessing topic manager", e);
         }
     }
 
+    /**
+     * Gets the JMS subscriptions for a topic
+     * Suppressing warning as this is used as a web service
+     *
+     * @param topic Topic name
+     * @return An array of subscriptions
+     * @throws EventAdminException
+     */
+    @SuppressWarnings("UnusedDeclaration")
     public Subscription[] getJMSSubscriptionsForTopic(String topic)
             throws EventAdminException {
         EventBroker eventBroker = EventAdminHolder.getInstance().getEventBroker();
         try {
             return adaptSubscriptions(eventBroker.getTopicManager().getJMSSubscriptions(topic));
         } catch (EventBrokerException e) {
-            throw new EventAdminException("Can not get the jms subscriptions", e);
+            log.error("Cannot get the jms subscriptions", e);
+            throw new EventAdminException("Cannot get the jms subscriptions", e);
         }
     }
 
-    private Subscription adaptSubscription(
-            org.wso2.carbon.event.core.subscription.Subscription coreSubscription) {
-        Calendar calendar = Calendar.getInstance();
-        calendar.setTime(coreSubscription.getCreatedTime());
-        Subscription adminSubscription = new Subscription();
-        adminSubscription.setCreatedTime(calendar);
-        adminSubscription.setEventDispatcher(coreSubscription.getEventDispatcher());
-        adminSubscription.setEventDispatcherName(coreSubscription.getEventDispatcherName());
-        adminSubscription.setEventFilter(coreSubscription.getEventFilter());
-        adminSubscription.setEventSinkURL(coreSubscription.getEventSinkURL());
-        adminSubscription.setExpires(coreSubscription.getExpires());
-        adminSubscription.setId(coreSubscription.getId());
-        adminSubscription.setOwner(coreSubscription.getOwner());
-        adminSubscription.setTopicName(coreSubscription.getTopicName());
-        adminSubscription.setMode(coreSubscription.getMode());
-        return adminSubscription;
+    /**
+     * Gets user roles through topic manager
+     *
+     * @return A string array of roles
+     * @throws EventAdminException
+     */
+    public String[] getUserRoles() throws EventAdminException {
+        EventBroker eventBroker = EventAdminHolder.getInstance().getEventBroker();
+        try {
+            return eventBroker.getTopicManager().getBackendRoles();
+        } catch (EventBrokerException e) {
+            log.error("Error in getting user roles from topic manager", e);
```

Comment:
The TopicManagerAdminServer is used by the UI. Logging of the error at the JSP page would require to use e.printStackTrace(). Which is not a good way of handling it. Therefore it is handled at the server layer. The logged error is throw to the JSP page where the JSP page would show an error pop up for the user.